Source/WTF/wtf/playstation/FileSystemPlayStation.cpp

```diff
@@ -99,9 +99,8 @@ Vector<String> listDirectorySub(const String& path, bool fullPath)
                 continue;
             String newEntry;
             if (fullPath) {
-                char filePath[PATH_MAX];
-                if (fullPath && static_cast<int>(sizeof(filePath) - 1) < snprintf(filePath, sizeof(filePath), "%s/%s", cpath.data(), name))
-                    continue; // buffer overflow
+                char filePath[strlen(cpath.data() + 1 + strlen(name) + 1];
```

Comment:
This is not fine as it is using variable-length stack array.

Reply:
Yes, this has to be fixed. Maybe the best fix is something like this:
```
newEntry = makeString(path, '/', stringFromFileSystemRepresentation(name));
```

Reply:
ok, done so